cpp/src/parquet/chunker_internal_test.cc

```diff
@@ -0,0 +1,1397 @@
+    result.push_back(
+        {{first.begin() + last_i, first.end()}, {second.begin() + last_j, second.end()}});
+  }
+
+  // Post-process: merge adjacent diffs to avoid splitting single changes into multiple
+  // parts
+  std::vector<ChunkDiff> merged;
+  for (auto& diff : result) {
+    if (!merged.empty()) {
+      auto& prev = merged.back();
+      // Check if we can merge with the previous diff
+      bool can_merge_a = prev.first.empty() && !prev.second.empty() &&
+                         !diff.first.empty() && diff.second.empty();
+      bool can_merge_b = prev.second.empty() && !prev.first.empty() &&
+                         !diff.second.empty() && diff.first.empty();
+
+      if (can_merge_a) {
+        // Combine into one diff: keep prev's second, use diff's first
+        prev.first = std::move(diff.first);
+        continue;
+      } else if (can_merge_b) {
+        // Combine into one diff: keep prev's first, use diff's second
+        prev.second = std::move(diff.second);
+        continue;
+      }
+    }
+    // If we can't merge, add this diff to the result
+    merged.push_back(std::move(diff));
+  }
+  return merged;
+}
+
+void PrintDifferences(const ChunkList& original, const ChunkList& modified,
+                      std::vector<ChunkDiff>& diffs) {
+  // Utility function to print the original and modified sequences, and the diffs
+  // between them. Used in case of failing assertions to display the differences.
+  std::cout << "Original: ";
+  for (const auto& val : original) {
+    std::cout << val << " ";
+  }
+  std::cout << std::endl;
+
+  std::cout << "Modified: ";
+  for (const auto& val : modified) {
+    std::cout << val << " ";
+  }
+  std::cout << std::endl;
+
+  for (const auto& diff : diffs) {
+    std::cout << "First: ";
+    for (const auto& val : diff.first) {
+      std::cout << val << " ";
+    }
+    std::cout << std::endl;
+
+    std::cout << "Second: ";
+    for (const auto& val : diff.second) {
+      std::cout << val << " ";
+    }
+    std::cout << std::endl;
+  }
+}
+
+TEST(TestFindDifferences, Basic) {
+  ChunkList first = {1, 2, 3, 4, 5};
+  ChunkList second = {1, 7, 8, 4, 5};
+
+  auto diffs = FindDifferences(first, second);
+
+  ASSERT_EQ(diffs.size(), 1);
+  ASSERT_EQ(diffs[0].first, ChunkList({2, 3}));
+  ASSERT_EQ(diffs[0].second, ChunkList({7, 8}));
+}
+
+TEST(TestFindDifferences, MultipleDifferences) {
+  ChunkList first = {1, 2, 3, 4, 5, 6, 7};
+  ChunkList second = {1, 8, 9, 4, 10, 6, 11};
+  auto diffs = FindDifferences(first, second);
+
+  ASSERT_EQ(diffs.size(), 3);
+
+  ASSERT_EQ(diffs[0].first, ChunkList({2, 3}));
+  ASSERT_EQ(diffs[0].second, ChunkList({8, 9}));
+
+  ASSERT_EQ(diffs[1].first, ChunkList({5}));
+  ASSERT_EQ(diffs[1].second, ChunkList({10}));
+
+  ASSERT_EQ(diffs[2].first, ChunkList({7}));
+  ASSERT_EQ(diffs[2].second, ChunkList({11}));
+}
+
+TEST(TestFindDifferences, DifferentLengths) {
+  ChunkList first = {1, 2, 3};
+  ChunkList second = {1, 2, 3, 4, 5};
```

Comment:
Can you make a slightly more complex example with several differences (including one at the start and one at the end, perhaps)?

Reply:
Yes, added.